controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -846,7 +861,15 @@ private TxnStatus handleDataNotFoundException(Throwable ex) {
                     byte[] updatedTable = TableHelper.addPartialRecordToHistoryTable(historyTable.getData(), newActiveSegments);
                     final Data<T> updated = new Data<>(updatedTable, historyTable.getVersion());
                     int latestEpoch = TableHelper.getLatestEpoch(updatedTable).getKey();
-                    return createNewEpoch(latestEpoch).thenCompose(v -> updateHistoryTable(updated));
+
+                    return createNewEpoch(latestEpoch).thenCompose(v -> updateHistoryTable(updated))
+                            .whenComplete((r, e) -> {
+                                if (e == null) {
+                                    log.debug("{}/{} scale op for epoch {}. Creating new epoch and updating history table.", scope, name, epoch);
+                                } else {
+                                    log.warn("{}/{} scale op for epoch {}. Failed to update history table. {}", scope, name, epoch, e.getClass().getName());
```

Comment:
we could use the following api org.slf4j.Logger#warn(java.lang.String, java.lang.Throwable)